src/auth.ts

```diff
@@ -425,6 +425,67 @@ function urlsafeBase64(base64string: string) {
   return base64string.replace(/\+/g, "-").replace(/=+$/, "").replace(/\//g, "_");
 }
 
+interface PrototyperRes {
+  uri: string;
+  sessionId: string;
+  authorize: (authorizationCode: string) => Promise<UserCredentials>;
+}
+
+export async function loginPrototyper(): Promise<PrototyperRes> {
+  const authProxyClient = new apiv2.Client({
+    urlPrefix: authProxyOrigin(),
```

Comment:
Seems like we call this multiple times, is this something we can just cache once and re-use?